lib/class-wp-theme-json-gutenberg.php

```diff
@@ -811,20 +811,46 @@ protected static function sanitize( $input, $valid_block_names, $valid_element_n
 	 *
 	 * @since 5.8.0
 	 * @since 6.1.0 Added append position.
+	 * @since 6.3.0 Deprecated append position parameter.
 	 *
-	 * @param string $selector  Original selector.
-	 * @param string $to_append Selector to append.
-	 * @param string $position  A position sub-selector should be appended. Default 'right'.
+	 * @param string $selector   Original selector.
+	 * @param string $to_append  Selector to append.
+	 * @param string $deprecated Deprecated parameter.
 	 * @return string The new selector.
 	 */
-	protected static function append_to_selector( $selector, $to_append, $position = 'right' ) {
+	protected static function append_to_selector( $selector, $to_append, $deprecated = 'right' ) {
 		if ( ! str_contains( $selector, ',' ) ) {
-			return 'right' === $position ? $selector . $to_append : $to_append . $selector;
+			return $selector . $to_append;
 		}
 		$new_selectors = array();
 		$selectors     = explode( ',', $selector );
 		foreach ( $selectors as $sel ) {
-			$new_selectors[] = 'right' === $position ? $sel . $to_append : $to_append . $sel;
+			$new_selectors[] = $sel . $to_append;
+		}
+		return implode( ',', $new_selectors );
+	}
+
+	/**
+	 * Prepends a sub-selector to an existing one.
+	 *
+	 * Given the compounded $selector "h1, h2, h3"
+	 * and the $to_prepend selector ".some-class " the result will be
+	 * ".some-class h1, .some-class h2, .some-class h3".
```

Comment:
As noted in a Slack DM to @felixarntz:

A `$to_prepend` of `".some-class "` will produce: `.some-class h1,.some-class  h2,.some-class  h3`.

This was the result prior to this PR though. See [this 3v4l](https://3v4l.org/VYbRZ).

One way to achieve the result in the docblock is `trim( $sel )` and `implode( ', ', $new_selectors )`. See [this 3v4l](https://3v4l.org/TdhDE).

I'm not sure if these "misplaced" spaces were already known and accepted, or if this is something tests should have caught.

Reply:
Thanks @costdev, good catch!

We could add `trim()` to the function, but to me that seems like a separate conversation. The output with `trim` would be preferable, but it's also not a big deal to have the extra space, and since the function is called tons of times there may be _some_ (probably insignificant) performance cost to it.

In other words, I'm by no means opposed, but potentially this could be discussed in a separate issue / PR.

I have fixed the doc block to include the double spaces to clarify that.

Reply:
Sounds like a plan @felixarntz!

As an alternative to the `trim()`, if we _know_ that the format always includes a space after the comma, we could `explode( ', ', $selector )` and `implode( ', ', $selector )`. See [this 3v4l](https://3v4l.org/QOaAV).